module4/lessons/Week4/Inheritance.md

```diff
@@ -0,0 +1,126 @@
+    {
+        Console.WriteLine("Woof! Woof!");
+    }
+}
+```
+
+Using Inherited Properties and Methods:
+With the inheritance relationship established, you can now create objects of the derived class and use both the properties and methods defined in the base class.
+
+```c#
+class Program
+{
+    static void Main()
+    {
+        Dog dog = new Dog
+        {
+            Species = "Canine",
+            Age = 3,
+            Breed = "Golden Retriever"
+        };
+
+        Console.WriteLine($"Species: {dog.Species}");
+        Console.WriteLine($"Age: {dog.Age}");
+        dog.Move(); // Inherited method from Animal class
+        dog.Bark(); // Method from the Dog class
+    }
+}
+```
+
+Output:
+```
+Species: Canine
+Age: 3
+Animal is moving.
+Woof! Woof!
+```
+
+In this example, the Dog class is a derived class inheriting from the Animal base class. The Dog class has its own property Breed and method Bark, while it also inherits the Species, Age, and MakeSound method from the Animal class.
+
+## Part 2: Pair Research on Inheritance
+
+The basics of inheritance in C# are almost certainly still confusing. Start by using Google to find 2-3 more resources introducing C# inheritance to read and deepen your understanding. Work together as a partnership, one of the best ways to learn a topic is by explaining. Try to explain what you are learning to each other!
+
+✅ CHECKPOINT ✅: Before continuing, work together to write a Slack message to your instructor explaining inheritance in your own words. Imagine your audience was you from last week and isn't familiar with the topic. Create your own example to use in the explanation. I recommend using a repl to code your example and ry things out.
+
+Once you get the go-ahead from your instructor, move on to thinking about and researching why this might be helpful.
+
+1. What are some of the benefits of using inheritance?
+2. What might be some of the disadvantages of using inheritance?
+
+Next, look for past examples of where you have used inheritance previously in this course.
+1. How is inheritance used here?
+1. Why is inheritance used here?
+
+✅ CHECKPOINT ✅: Before continuing, work together to write a Slack message to your instructor explaining the example of inheritance that you found. Be sure to include how and why inheritance is used here.
+
+Finally, move on to digging into a couple more advanced topics. Here are some ideas, or feel free to dig into your own questions!
+
+1. In C# can you inherit from multiple classes or just one?
+1. How does C# handle constructors in inherited classes?
+1. Can I prevent a class from being inherited by other classes?
+1. What are access modifiers in inheritance (public, protected, private, etc.), and how do they affect inherited members?
+1. What happens if you define the same method in the parent class and the child class?
+1. What is method overriding?
+1. What are abstract classes and when should I use them?
+
+## Part 3: Presentation Requirements

```

Comment:
Are we thinking this will be a presentation to the whole cohort? or will we split into smaller groups?  If possible, I think smaller groups works better because the info can get a bit repetitive.

Reply:
I'm thinking whole group for this bunch. With only 10 students and working in pairs, it's less than an hour of presentations for two days. And I think there will be benefits from seeing how other students present similar topics. I tried to structure it so there is also some differences in what MVC example and what advanced topic they present on so there is learning from each other too.

I absolutely think we should split it up for larger cohorts, but for 10 students I'm thinking one group. Does that sound alright to you?